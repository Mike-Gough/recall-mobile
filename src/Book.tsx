import React from 'react';
import { StyleSheet, Text, View, TouchableHighlight } from 'react-native';
import { NavigationScreenProps, NavigationInjectedProps, FlatList } from 'react-navigation';
import _ from 'lodash';

export interface BookDetails {
  id: number;
  title: string;
}

export interface BookScreenProps {
  book: string;
  details: Array<BookDetails>;
}

export default class Book extends React.Component<NavigationScreenProps<BookScreenProps>> {
  static navigationOptions = ({ navigation }: NavigationScreenProps<BookScreenProps>) => ({
    title: `${navigation.state.params ? navigation.state.params.book : 'Book'}`,
  });

  render() {
    const props: NavigationInjectedProps<BookScreenProps> = this.props;
    const book: string =
      (props.navigation.state.params && props.navigation.state.params.book) || 'unknown';
    const entries: Array<BookDetails> | undefined =
      (props.navigation.state.params && props.navigation.state.params.details) || [];
    let renderSeparator = () => {
      return (
        <View
          style={{
            height: 1,
            width: '100%',
            backgroundColor: '#CED0CE',
            marginLeft: 15,
          }}
        />
      );
    };

    let titleToHeading = (_title: string) => {
      return _title;
    };

    let titleToNumberOfLines = (_title: string) => {
      return 1;
    };

    let noteClicked = (_note: BookDetails) => () => {
      this.props.navigation.navigate('Note', _note);
    };

    let extractKey = (item: BookDetails, index: number) => {
      return item.id.toString();
    };

    let renderItem = ({ item }: { item: BookDetails }) => {
      return (
        <TouchableHighlight onPress={noteClicked(item)}>
          <View style={styles.itemContainer}>
            <Text style={styles.item} numberOfLines={titleToNumberOfLines(item.title)}>
              {titleToHeading(item.title)}
            </Text>
          </View>
        </TouchableHighlight>
      );
    };

    return (
      <View style={styles.container}>
        <FlatList
          ItemSeparatorComponent={renderSeparator}
          data={entries}
          keyExtractor={extractKey}
          renderItem={renderItem}
        />
      </View>
    );
  }
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: 'white',
    alignItems: 'stretch',
    justifyContent: 'space-evenly',
  },
  itemContainer: {
    flexWrap: 'wrap',
    alignItems: 'flex-start',
    flexDirection: 'row',
  },
  item: {
    color: 'black',
    padding: 15,
    fontSize: 18,
  },
  itemIcon: {
    paddingLeft: 15,
    paddingTop: 16,
    paddingBottom: 15,
    paddingRight: 15,
  },
});
