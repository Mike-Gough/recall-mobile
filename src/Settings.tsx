import React from 'react';
import { StyleSheet, Text, View, Button } from 'react-native';
import { NavigationScreenProps, NavigationStackScreenOptions } from 'react-navigation';

export default class Friends extends React.Component<NavigationScreenProps> {
  static navigationOptions = (screenProps: NavigationScreenProps): NavigationStackScreenOptions => {
    return {
      title: 'Settings',
    };
  };

  private homeButtonClicked = () => {
    this.props.navigation.navigate('Notes');
  };

  render() {
    return (
      <View style={styles.container}>
        <Text>Settings Page!</Text>
        <Button title="Back to notes" onPress={this.homeButtonClicked} />
      </View>
    );
  }
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: 'white',
    alignItems: 'center',
    justifyContent: 'center',
    padding: 10,
  },
});
