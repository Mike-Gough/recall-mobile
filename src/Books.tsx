import React from 'react';
import { StyleSheet, FlatList, Text, View, TouchableHighlight } from 'react-native';
import { NavigationScreenProps } from 'react-navigation';
import Ionicons from 'react-native-vector-icons/Ionicons';
import { BookScreenProps } from './Book';

export default class Books extends React.Component<NavigationScreenProps> {
  static navigationOptions = {
    title: 'Books',
  };

  static books: Array<BookScreenProps> = [
    {
      book: 'JavaScript',
      details: [
        {
          id: 1,
          title: '<javascript example>',
        },
        {
          id: 2,
          title: `\`await\` suspends the current *function* evaluations, including all control structures. 
                    
                    therefore in \`.forEach\` it does not wait for callback to finish before iterating to next.
                    
                    https://stackoverflow.com/questions/37576685/using-async-await-with-a-foreach-loop`,
        },
        {
          id: 3,
          title: `# This is Heading 1{'\n\n'}
                            ## This is Heading 2{'\n\n'}
                            1. List1 {'\n'}
                            2. List2 {'\n'} 
                            {'\t'}1. sublist1{'\n'}
                            {'\t'}description sublist1.{'\n'}
                            {'\t'}2. sublist2{'\t\t'}{'\n'}
                            * List3{'\n'}
                            * List4{'\n'}
                            {'\t'}1. **bold text** {'\n'}
                            {'\t'}2. this line contain * and should not be a new sublist{'\t\t'}{'\n'}
                            5.  Last list{'\n\n'}
                            Some *really* basic **Markdown**.{'\n\n'}


                            ## this is header {'\n\n\n\n'}
                            {'this is _italic_ '}
                            {'this is **strong**'}

                            {'\n\n'}
                            | # | Name   | Age |{'\n'}
                            |---|--------|-----|{'\n'}
                            | 1 | John   | 19  |{'\n'}
                            | 2 | Sally  | 18  |{'\n'}
                            | 3 | Stream | 20  |{'\n'}`,
        },
      ],
    },
    {
      book: 'Linux',
      details: [
        {
          id: 1,
          title: 'linux example',
        },
      ],
    },
    {
      book: 'Wisdom',
      details: [
        {
          id: 1,
          title: 'wisodom example',
        },
      ],
    },
    {
      book: 'Vocabulary',
      details: [
        {
          id: 1,
          title: 'vocabulary example',
        },
      ],
    },
  ];

  render() {
    let tint: string | undefined = undefined;
    let IconComponent = Ionicons;
    let iconName = `ios-paper`;
    let renderSeparator = () => {
      return (
        <View
          style={{
            height: 1,
            width: '100%',
            backgroundColor: '#CED0CE',
            marginLeft: 15,
          }}
        />
      );
    };

    let extractKey = (item: BookScreenProps, index: number) => {
      return item.book.toString();
    };

    let bookClicked = (_book: BookScreenProps) => () =>
      this.props.navigation.navigate('Book', _book);

    let renderItem = ({ item }: { item: BookScreenProps }) => {
      return (
        <TouchableHighlight onPress={bookClicked(item)}>
          <View style={styles.itemContainer}>
            <IconComponent style={styles.itemIcon} name={iconName} size={18} color={tint} />
            <Text style={styles.item}>{item.book}</Text>
          </View>
        </TouchableHighlight>
      );
    };

    return (
      <View style={styles.container}>
        <FlatList
          ItemSeparatorComponent={renderSeparator}
          data={Books.books}
          keyExtractor={extractKey}
          renderItem={renderItem}
        />
      </View>
    );
  }
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: 'white',
    alignItems: 'stretch',
    justifyContent: 'space-evenly',
  },
  itemContainer: {
    flexWrap: 'wrap',
    alignItems: 'flex-start',
    flexDirection: 'row',
  },
  item: {
    color: 'black',
    fontWeight: 'bold',
    paddingTop: 15,
    paddingBottom: 15,
    fontSize: 16,
  },
  itemIcon: {
    paddingLeft: 15,
    paddingTop: 16,
    paddingBottom: 12,
    paddingRight: 15,
  },
});
