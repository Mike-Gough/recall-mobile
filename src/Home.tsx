import React from 'react';
import { StyleSheet, Button, Text, View } from 'react-native';
import { NavigationScreenProps, NavigationStackScreenOptions } from 'react-navigation';
import { BookScreenProps } from './Book';

export default class Home extends React.Component<NavigationScreenProps> {
  static navigationOptions = (screenProps: NavigationScreenProps): NavigationStackScreenOptions => {
    return {
      title: 'Notes',
    };
  };

  private homeButtonClicked = () => {
    this.props.navigation.navigate('Books');
  };

  render() {
    return (
      <View style={styles.container}>
        <Text>Notes Page!</Text>
        <Button title="Go to books" onPress={this.homeButtonClicked} />
      </View>
    );
  }
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: 'white',
    alignItems: 'center',
    justifyContent: 'center',
    padding: 10,
  },
});
