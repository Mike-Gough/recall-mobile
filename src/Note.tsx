import React from 'react';
import { StyleSheet, ScrollView } from 'react-native';
import { NavigationScreenProps, NavigationInjectedProps } from 'react-navigation';
import _ from 'lodash';
import { BookDetails } from './Book';
import Markdown from 'react-native-markdown-renderer';

export default class Note extends React.Component<NavigationScreenProps<BookDetails>> {
  static navigationOptions = ({ navigation }: NavigationScreenProps<BookDetails>) => ({
    title: `${navigation.state.params ? navigation.state.params.title : 'Note'}`,
  });

  render() {
    const props: NavigationInjectedProps<BookDetails> = this.props;
    const title: string =
      (props.navigation.state.params && props.navigation.state.params.title) || 'unknown';

    return (
      <ScrollView style={styles.container}>
        <Markdown style={markdownStyle}>{title}</Markdown>
      </ScrollView>
    );
  }
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: 'white',
    padding: 10,
    position: 'absolute',
    top: 0,
    height: '100%',
    width: '100%',
  },
  itemContainer: {
    flexWrap: 'wrap',
    alignItems: 'flex-start',
    flexDirection: 'row',
  },
});

const markdownStyle = StyleSheet.create({
  view: {
    height: '100%',
  },
  heading: {
    borderBottomWidth: 1,
    borderColor: '#000000',
  },
  heading1: {
    fontSize: 32,
    backgroundColor: '#000000',
    color: '#FFFFFF',
  },
  heading2: {
    fontSize: 24,
  },
  heading3: {
    fontSize: 18,
  },
  heading4: {
    fontSize: 16,
  },
  heading5: {
    fontSize: 13,
  },
  heading6: {
    fontSize: 11,
  },
});
