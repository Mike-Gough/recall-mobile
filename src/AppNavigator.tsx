import React from 'react';
import {
  createBottomTabNavigator,
  createStackNavigator,
  NavigationStackScreenOptions,
} from 'react-navigation';
import Home from './Home';
import Books from './Books';
import Book from './Book';
import Note from './Note';
import Settings from './Settings';
import Ionicons from 'react-native-vector-icons/Ionicons';

let navigationOptions: NavigationStackScreenOptions = {
  headerTitleStyle: {
    fontSize: 22,
    textAlign: 'left',
    alignSelf: 'auto',
  },
  headerStyle: {
    backgroundColor: 'white',
  },
};

const NotesStack = createStackNavigator(
  {
    Notes: Home,
  },
  {
    defaultNavigationOptions: navigationOptions,
  },
);

const BooksStack = createStackNavigator(
  {
    Books: Books,
    Book: Book,
    Note: Note,
  },
  {
    defaultNavigationOptions: navigationOptions,
  },
);

const SettingsStack = createStackNavigator(
  {
    Settings: Settings,
  },
  {
    defaultNavigationOptions: navigationOptions,
  },
);

const TabNavigator = createBottomTabNavigator(
  {
    Notes: NotesStack,
    Books: BooksStack,
    Settings: SettingsStack,
  },
  {
    defaultNavigationOptions: ({ navigation }) => ({
      tabBarIcon: ({ focused, horizontal, tintColor }) => {
        const { routeName } = navigation.state;
        let tint = tintColor ? tintColor : undefined;
        let IconComponent = Ionicons;
        let iconName;

        switch (routeName) {
          case 'Notes': {
            iconName = `ios-paper`;
            break;
          }
          case 'Books': {
            iconName = `ios-journal`;
            break;
          }
          default: {
            iconName = `ios-options`;
            break;
          }
        }

        return <IconComponent name={iconName} size={25} color={tint} />;
      },
    }),
    tabBarOptions: {
      activeTintColor: 'tomato',
      inactiveTintColor: 'gray',
    },
  },
);

const AppNavigator = createStackNavigator(
  {
    Home: TabNavigator,
  },
  {
    headerMode: 'none',
  },
);

export default AppNavigator;
