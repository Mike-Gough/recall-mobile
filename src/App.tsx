import React from 'react';
import AppNavigator from './AppNavigator';
import { createAppContainer, createBottomTabNavigator } from 'react-navigation';


const AppContainer = createAppContainer(AppNavigator);

export default class App extends React.Component {
  render() {
    return <AppContainer />;
  }
}