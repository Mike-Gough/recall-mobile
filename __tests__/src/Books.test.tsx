import React from 'react';
import Books from '../../src/Books';
import renderer from 'react-test-renderer';
import { NavigationStackScreenOptions, NavigationScreenProp, NavigationRoute, NavigationParams } from 'react-navigation';
import { BookScreenProps } from '../../src/Book';

test('Render successfully', () => {

    let navigationOptions: NavigationStackScreenOptions = {
        title: 'Books'
    }

    let props: any;

    let navigation: NavigationScreenProp<NavigationRoute<NavigationParams>, NavigationParams> = {
        dispatch: props,
        goBack: props,
        dismiss: props,
        navigate: props,
        openDrawer: props,
        closeDrawer: props,
        toggleDrawer: props,
        getParam: props,
        setParams: props,
        addListener: props,
        push: props,
        replace: props,
        pop: props,
        popToTop: props,
        isFocused: props,
        state: props,
        dangerouslyGetParent: props

    }

    const component = renderer.create(
        <Books navigation={navigation} navigationOptions={navigationOptions} />
    );

    let generatedComponent = component.toJSON();
    expect(generatedComponent).toMatchSnapshot();
});