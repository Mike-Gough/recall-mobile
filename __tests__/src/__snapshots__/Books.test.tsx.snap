// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render successfully 1`] = `
<View
  style={
    Object {
      "alignItems": "stretch",
      "backgroundColor": "white",
      "flex": 1,
      "justifyContent": "space-evenly",
    }
  }
>
  <RCTScrollView
    ItemSeparatorComponent={[Function]}
    data={
      Array [
        Object {
          "book": "JavaScript",
          "details": Array [
            Object {
              "id": 1,
              "title": "<javascript example>",
            },
            Object {
              "id": 2,
              "title": "\`await\` suspends the current *function* evaluations, including all control structures. 
                    
                    therefore in \`.forEach\` it does not wait for callback to finish before iterating to next.
                    
                    https://stackoverflow.com/questions/37576685/using-async-await-with-a-foreach-loop",
            },
            Object {
              "id": 3,
              "title": "# This is Heading 1{'

'}
                            ## This is Heading 2{'

'}
                            1. List1 {'
'}
                            2. List2 {'
'} 
                            {'	'}1. sublist1{'
'}
                            {'	'}description sublist1.{'
'}
                            {'	'}2. sublist2{'		'}{'
'}
                            * List3{'
'}
                            * List4{'
'}
                            {'	'}1. **bold text** {'
'}
                            {'	'}2. this line contain * and should not be a new sublist{'		'}{'
'}
                            5.  Last list{'

'}
                            Some *really* basic **Markdown**.{'

'}


                            ## this is header {'



'}
                            {'this is _italic_ '}
                            {'this is **strong**'}

                            {'

'}
                            | # | Name   | Age |{'
'}
                            |---|--------|-----|{'
'}
                            | 1 | John   | 19  |{'
'}
                            | 2 | Sally  | 18  |{'
'}
                            | 3 | Stream | 20  |{'
'}",
            },
          ],
        },
        Object {
          "book": "Linux",
          "details": Array [
            Object {
              "id": 1,
              "title": "linux example",
            },
          ],
        },
        Object {
          "book": "Wisdom",
          "details": Array [
            Object {
              "id": 1,
              "title": "wisodom example",
            },
          ],
        },
        Object {
          "book": "Vocabulary",
          "details": Array [
            Object {
              "id": 1,
              "title": "vocabulary example",
            },
          ],
        },
      ]
    }
    disableVirtualization={false}
    getItem={[Function]}
    getItemCount={[Function]}
    horizontal={false}
    initialNumToRender={10}
    keyExtractor={[Function]}
    maxToRenderPerBatch={10}
    numColumns={1}
    onContentSizeChange={[Function]}
    onEndReachedThreshold={2}
    onLayout={[Function]}
    onMomentumScrollEnd={[Function]}
    onScroll={[Function]}
    onScrollBeginDrag={[Function]}
    onScrollEndDrag={[Function]}
    removeClippedSubviews={false}
    renderItem={[Function]}
    scrollEventThrottle={50}
    stickyHeaderIndices={Array []}
    updateCellsBatchingPeriod={50}
    viewabilityConfigCallbackPairs={Array []}
    windowSize={21}
  >
    <View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <View
          accessible={true}
          isTVSelectable={true}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={null}
        >
          <View
            style={
              Object {
                "alignItems": "flex-start",
                "flexDirection": "row",
                "flexWrap": "wrap",
              }
            }
          >
            <Text
              allowFontScaling={false}
              style={
                Array [
                  Object {
                    "color": undefined,
                    "fontSize": 18,
                  },
                  Object {
                    "paddingBottom": 12,
                    "paddingLeft": 15,
                    "paddingRight": 15,
                    "paddingTop": 16,
                  },
                  Object {
                    "fontFamily": "Ionicons",
                    "fontStyle": "normal",
                    "fontWeight": "normal",
                  },
                  Object {},
                ]
              }
            >
              
            </Text>
            <Text
              style={
                Object {
                  "color": "black",
                  "fontSize": 16,
                  "fontWeight": "bold",
                  "paddingBottom": 15,
                  "paddingTop": 15,
                }
              }
            >
              JavaScript
            </Text>
          </View>
        </View>
        <View
          style={
            Object {
              "backgroundColor": "#CED0CE",
              "height": 1,
              "marginLeft": 15,
              "width": "100%",
            }
          }
        />
      </View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <View
          accessible={true}
          isTVSelectable={true}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={null}
        >
          <View
            style={
              Object {
                "alignItems": "flex-start",
                "flexDirection": "row",
                "flexWrap": "wrap",
              }
            }
          >
            <Text
              allowFontScaling={false}
              style={
                Array [
                  Object {
                    "color": undefined,
                    "fontSize": 18,
                  },
                  Object {
                    "paddingBottom": 12,
                    "paddingLeft": 15,
                    "paddingRight": 15,
                    "paddingTop": 16,
                  },
                  Object {
                    "fontFamily": "Ionicons",
                    "fontStyle": "normal",
                    "fontWeight": "normal",
                  },
                  Object {},
                ]
              }
            >
              
            </Text>
            <Text
              style={
                Object {
                  "color": "black",
                  "fontSize": 16,
                  "fontWeight": "bold",
                  "paddingBottom": 15,
                  "paddingTop": 15,
                }
              }
            >
              Linux
            </Text>
          </View>
        </View>
        <View
          style={
            Object {
              "backgroundColor": "#CED0CE",
              "height": 1,
              "marginLeft": 15,
              "width": "100%",
            }
          }
        />
      </View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <View
          accessible={true}
          isTVSelectable={true}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={null}
        >
          <View
            style={
              Object {
                "alignItems": "flex-start",
                "flexDirection": "row",
                "flexWrap": "wrap",
              }
            }
          >
            <Text
              allowFontScaling={false}
              style={
                Array [
                  Object {
                    "color": undefined,
                    "fontSize": 18,
                  },
                  Object {
                    "paddingBottom": 12,
                    "paddingLeft": 15,
                    "paddingRight": 15,
                    "paddingTop": 16,
                  },
                  Object {
                    "fontFamily": "Ionicons",
                    "fontStyle": "normal",
                    "fontWeight": "normal",
                  },
                  Object {},
                ]
              }
            >
              
            </Text>
            <Text
              style={
                Object {
                  "color": "black",
                  "fontSize": 16,
                  "fontWeight": "bold",
                  "paddingBottom": 15,
                  "paddingTop": 15,
                }
              }
            >
              Wisdom
            </Text>
          </View>
        </View>
        <View
          style={
            Object {
              "backgroundColor": "#CED0CE",
              "height": 1,
              "marginLeft": 15,
              "width": "100%",
            }
          }
        />
      </View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <View
          accessible={true}
          isTVSelectable={true}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={null}
        >
          <View
            style={
              Object {
                "alignItems": "flex-start",
                "flexDirection": "row",
                "flexWrap": "wrap",
              }
            }
          >
            <Text
              allowFontScaling={false}
              style={
                Array [
                  Object {
                    "color": undefined,
                    "fontSize": 18,
                  },
                  Object {
                    "paddingBottom": 12,
                    "paddingLeft": 15,
                    "paddingRight": 15,
                    "paddingTop": 16,
                  },
                  Object {
                    "fontFamily": "Ionicons",
                    "fontStyle": "normal",
                    "fontWeight": "normal",
                  },
                  Object {},
                ]
              }
            >
              
            </Text>
            <Text
              style={
                Object {
                  "color": "black",
                  "fontSize": 16,
                  "fontWeight": "bold",
                  "paddingBottom": 15,
                  "paddingTop": 15,
                }
              }
            >
              Vocabulary
            </Text>
          </View>
        </View>
      </View>
    </View>
  </RCTScrollView>
</View>
`;
